pallets/file-system/src/utils.rs

```diff
@@ -482,11 +497,15 @@ where
             &file_proof,
         )?;
 
+        // Get the file metadata to insert into the bucket under the file key.
+        let file_metadata = storage_request_metadata.clone().to_file_metadata();
+        let encoded_trie_value = file_metadata.encode();
```

Comment:
We should think how to be careful to always encode consistently.

Reply:
As long as we use the FileMetadata type, we will always encode and decode correctly since that's the type that derives the Encode and Decode traits. Do you have any other safeguards in mind that we could implement?